exercises/src/main/scala/fpinscala/parallelism/Par.scala

```diff
@@ -63,4 +147,40 @@ object Examples {
       sum(l) + sum(r) // Recursively sum both halves and add the results together.
     }
 
+  def parSum(ints: IndexedSeq[Int]): Par[Int] =
+    if (ints.length <= 1)
+      Par.unit(ints.headOption getOrElse 0)
+    else {
+      val (l,r) = ints.splitAt(ints.length/2)
+      Par.map2(Par.fork(parSum(l)), Par.fork(parSum(r)))(_ + _)
+    }
+
+  def parIntOp(ints: IndexedSeq[Int])(f: (Int, Int) => Int): Par[Int] =
+    if (ints.length <= 1)
+      Par.unit(ints.headOption getOrElse 0)
```

Comment:
Size 1 and 0 are the base case used for generating the values in the sequence for which we can then apply `f`. So handling of 0 and 1 are definitely not an error. However, there could exist operations where we want to control the default value `0` but passing in a `z` parameter. For `parSum` `z` would be `0` and for `parMultiply` it would be `1`. WDYT?